torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -686,6 +686,34 @@ bool ProcessGroupNCCL::WorkNCCL::checkTimeout(
       " milliseconds before timing out.");
 
   LOG(ERROR) << exceptionMsg;
+
+  // Get the stack trace of the work at call time
+  // First step we get the corresponding record entry from FR, based on work's
+  // trace_id_
+  std::optional<NCCLTraceBuffer::Entry> entry =
```

Comment:
Could you place this piece of code into a helper function in WorkNCCL such that we can print the traceback whenever needed

Reply:
Yep, I was thinking about that too. In a next PR I will rearrange things. The current is kept as-is for "review sanity".